### ISSUE TITLE
How to update ESP8266 library for Sparkfun add-on

### ISSUE THREAD

**Issue by hanaldo** [sparkfun/Blynk_Board_ESP8266#10]
Not sure where I can ask this issue but I was recommended to post it here. In order to program the Sparkfun Blynk Board I installed the [add-on](https://raw.githubusercontent.com/sparkfun/Arduino_Boards/master/IDE_Board_Manager/package_sparkfun_index.json), but this add-on is using Blynk library v0.3.1 and ESP8266 library v2.1.0, I really do wish to update the ESP8266 library to its newest version due to some bugs and incompatibility issue to newest Blynk library. But I couldn't do so with the add-on, does anyone know how to update the ESP8266 core for Sparkfun add-on or is there any other way we can program the Sparkfun Blynk Board without its add-on?

**Comment by hanaldo** [comment]
It looks like the Sparkfun Blynk board [add-on](https://raw.githubusercontent.com/sparkfun/Arduino_Boards/master/IDE_Board_Manager/package_sparkfun_index.json) was updated, it is now using ESP8266 core v2.1.2 instead of v2.1.0, but it is actually broken. After I install the add-on it misses a lot of necessary non-ESP8266 libraries, such as Blynk library, SparkFun_HTU21D, SparkFun_TSL2561, then I have to install them all manually. Finally, the ESP8266 core v2.1.2 core can only compile with Blynk v0.5.2 and below, I was forced to dig into this myself. Btw, the [document](https://cdn.sparkfun.com/assets/learn_tutorials/4/9/5/Blynk-_Community_ESP8266_Boards.pdf) from Sparkfun for troubleshooting ESP8266 community board does not work at all!

**Comment by hanaldo** [comment]
I wanted to use the most up-to-date Blynk library, which is now v0.6.1. So, I manually updated my Blynk library to v0.6.1, and it is not compatible with ESP8266 core v2.1.2, then I manually updated my ESP8266 core to v2.5.2, and updated its "boards.txt" file to include my Sparkfun Blynk Board configuration. It can compile and run, but the board will throw an error of "ISR not in IRAM!" when running the Sparkfun's firmware. According to this [post](https://community.blynk.cc/t/error-isr-not-in-iram/37426/10), I downgraded my ESP8266 core to v2.4.2, and now everything works fine. In short, now I can flash my Sparkfun Blynk Board with Blynk library v0.6.1 and ESP8266 core v2.4.2!
